pep_sphinx_extensions/pep_theme/static/colour_scheme.js

```diff
@@ -1,36 +1,40 @@
-const dark = document.getElementById("css-dark");
-const pygmentsNormal = document.getElementById("pyg");
-const pygmentsDark = document.getElementById("pyg-dark");
-
-const makeLight = () => {
-    dark.disabled = pygmentsDark.disabled = true
-    dark.media = pygmentsNormal.media = pygmentsDark.media = ""
-    pygmentsNormal.disabled = false
+const prefersDark = window.matchMedia("(prefers-color-scheme: dark)")
 
+const getColourScheme = () => document.documentElement.dataset.colour_scheme
+const setColourScheme = (colourScheme = getColourScheme()) => {
+    document.documentElement.dataset.colour_scheme = colourScheme
+    setPygments(colourScheme)
+    setTooltip(nextColourScheme(colourScheme))
+    localStorage.setItem("colour_scheme", colourScheme)
 }
 
-const makeDark = () => {
-    dark.disabled = pygmentsDark.disabled = false
-    dark.media = pygmentsNormal.media = pygmentsDark.media = ""
-    pygmentsNormal.disabled = true
+// Map system theme to a cycle of steps
+const cycles = {
+    dark: ["auto", "light", "dark"], // auto (dark) → light → dark
+    light: ["auto", "dark", "light"], // auto (light) → dark → light
 }
 
+const nextColourScheme = (colourScheme = getColourScheme()) => {
+    const cycle = cycles[prefersDark.matches ? "dark" : "light"]
+    return cycle[(cycle.indexOf(colourScheme) + 1) % cycle.length]
+}
 
-const toggleColourScheme = () => {
-    const colourScheme = localStorage.getItem("colour_scheme")
-    const prefersDark = window.matchMedia("(prefers-color-scheme: dark)").matches
+const setPygments = (colourScheme = getColourScheme()) => {
+    const pygmentsDark = document.getElementById("pyg-dark")
+    const pygmentsLight = document.getElementById("pyg-light")
+    pygmentsDark.disabled = colourScheme === "light"
+    pygmentsLight.disabled = colourScheme === "dark"
+    pygmentsDark.media = colourScheme === "auto" ? "(prefers-color-scheme: dark)" : ""
+    pygmentsLight.media = colourScheme === "auto" ? "(prefers-color-scheme: light)" : ""
+}
 
-    if ((colourScheme === "dark") || (!colourScheme && prefersDark)) {
-        makeLight()
-        localStorage.setItem("colour_scheme", "light")
-    } else {
-        makeDark()
-        localStorage.setItem("colour_scheme", "dark")
-    }
+const setTooltip = (schemeOnClick = nextColourScheme()) => {
+    const label = schemeOnClick === "auto" ? "Adapt to system theme" : `Switch to ${schemeOnClick} mode`
+    const button = document.getElementById("colour-scheme-cycler")
+    button.setAttribute( "aria-label", label)
+    button.setAttribute( "title", label)
```

Comment:
```suggestion
    button.setAttribute("aria-label", label)
    button.setAttribute("title", label)
```

Small nit, but any reason for this space?